File: spring-boot-3-demo-app/src/test/java/io/github/mfvanek/spring/boot3/test/controllers/HomeControllerTest.java
/*
 * Copyright (c) 2020-2025. Ivan Vakhrushev and others.
 * https://github.com/mfvanek/spring-boot-open-telemetry-demo
 *
 * Licensed under the Apache License 2.0
 */

package io.github.mfvanek.spring.boot3.test.controllers;

import io.github.mfvanek.spring.boot3.test.support.TestBase;
import org.junit.jupiter.api.Test;

import static io.github.mfvanek.spring.boot3.test.filters.TraceIdInResponseServletFilter.TRACE_ID_HEADER_NAME;
import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

class HomeControllerTest extends TestBase {

    @Test
    void homeControllerShouldWork() {
        final String result = webTestClient.get()
            .uri("/")
            .exchange()
            .expectStatus().isEqualTo(200)
            .expectHeader().exists(TRACE_ID_HEADER_NAME)
            .expectBody(String.class)
            .returnResult()
            .getResponseBody();
        assertThat(result).isEqualTo("Hello!");
    }
}

File: spring-boot-3-demo-app-kotlin/src/test/kotlin/io/github/mfvanek/spring/boot3/kotlin/test/controllers/HomeControllerTest.kt
package io.github.mfvanek.spring.boot3.kotlin.test.controllers

import io.github.mfvanek.spring.boot3.kotlin.test.filters.TraceIdInResponseServletFilter.Companion.TRACE_ID_HEADER_NAME
import io.github.mfvanek.spring.boot3.kotlin.test.support.TestBase
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class HomeControllerTest : TestBase() {
    @Test
    fun homeControllerShouldWork() {
        val result = webTestClient.get()
            .uri("/")
            .exchange()
            .expectStatus().isEqualTo(200)
            .expectHeader().exists(TRACE_ID_HEADER_NAME)
            .expectBody(String::class.java)
            .returnResult()
            .responseBody
        assertThat<String>(result)
            .isEqualTo("Hello!")
    }
}
